azure_functions_worker/dispatcher.py

```diff
@@ -556,6 +553,19 @@ async def _handle__close_shared_memory_resources_request(self, req):
                 request_id=self.request_id,
                 close_shared_memory_resources_response=response)
 
+    @staticmethod
+    def _get_trace_and_retry_context(invocation_request):
```

Comment:
Is there any check needed if any one of the contexts is None? Why set an object as None itself?

Reply:
On second thought - can the whole Context object be made here? Bring in this code, within this function?
```
bindings.Context(
                fi.name, fi.directory, invocation_id,
                trace_context, retry_context)
```

You already have all the contexts. Also, get the context closer to where it's used - near L391?